bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,577 @@
+
+        if (comms == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' not completely initialized", label));
+            logger.debug("Bridge not initialized fully (no endpoint) -- aborting init for {}", this);
+            return;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(HeliosEasyControlsConfiguration.class);
+        this.readVariableDefinition();
+        this.connectEndpoint();
+        if (this.comms != null) {
+            HeliosEasyControlsHandler.transactionLocks.putIfAbsent(this.comms.getEndpoint(), new Semaphore(1, true));
+            updateStatus(ThingStatus.UNKNOWN);
+
+            // background initialization
+            scheduler.execute(() -> {
+                try {
+                    readValue(HeliosEasyControlsBindingConstants.DATE_FORMAT);
+                    // status will be updated to ONLINE by the read callback function
+                } catch (HeliosException e) {
+                    logger.error(e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                }
+            });
+
+            // poll for status updates regularly
+            Runnable runnable = new Runnable() {
+
+                @Override
+                public void run() {
+                    for (Map.Entry<String, HeliosVariable> entry : vMap.entrySet()) {
+                        if (isLinked(entry.getKey())
+                                || (entry.getKey().equals(HeliosEasyControlsBindingConstants.DATE_FORMAT))) {
+                            try {
+                                readValue(entry.getKey());
+                            } catch (HeliosException e) {
+                                logger.error(e.getMessage());
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        e.getMessage());
+                            }
+                        }
+                    }
+                }
+            };
+            this.pollingJob = scheduler.scheduleAtFixedRate(runnable, config.getRefreshInterval(),
+                    config.getRefreshInterval(), TimeUnit.MILLISECONDS);
+
+        } else {
+            logger.error("Modbus endpoint is null.");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (this.pollingJob != null) {
+            this.pollingJob.cancel(true);
+        }
+        this.comms = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            scheduler.schedule(new Runnable() {

```

Comment:
scheduler.submit is a cleaner way to schedule a runnable to be executed immediately on the background, you can drop the other arguments